common_benchmark_suite/openxla/benchmark/models/jax/generate_model_artifacts.py

```diff
@@ -0,0 +1,181 @@
+import tarfile
+
+from typing import Any, Optional
+
+# Add openxla dir to the search path.
+sys.path.insert(0, str(pathlib.Path(__file__).parents[4]))
+
+from openxla.benchmark import def_types
+from openxla.benchmark.comparative_suite.jax import model_definitions
+from openxla.benchmark.models.jax import model_interfaces
+
+HLO_FILENAME_REGEX = r".*jit_forward.before_optimizations.txt"
+HLO_STATIC_FILENAME = "xla_hlo_before_optimizations.txt"
+
+
+def _generate_and_save_inputs(model_obj: Any,
+                              model_dir: str) -> tuple[Any, ...]:
+  raw_inputs = model_obj.generate_default_inputs()
+  inputs = model_obj.preprocess(raw_inputs)
+
+  # Save inputs.
+  inputs_dir = os.path.join(model_dir, "inputs")
+  os.makedirs(inputs_dir, exist_ok=True)
+  for idx, input in enumerate(inputs):
+    input_path = os.path.join(inputs_dir, f"input_{idx}.npy")
+    print(f"Saving input {jnp.shape(input)} to {input_path}")
+    jnp.save(input_path, input)
+
+  with tarfile.open(os.path.join(model_dir, "inputs_npy.tgz"), "w:gz") as tar:
+    tar.add(f"{inputs_dir}/", arcname=".")
+  shutil.rmtree(inputs_dir)
+
+  return inputs
+
+
+def _save_outputs(outputs: tuple[Any, ...], model_dir: str):
+  outputs_dir = os.path.join(model_dir, "outputs")
+  os.makedirs(outputs_dir, exist_ok=True)
+
+  for idx, output in enumerate(outputs):
+    output_path = os.path.join(outputs_dir, f"output_{idx}.npy")
+    print(f"Saving output {jnp.shape(outputs)} to {output_path}")
+    jnp.save(output_path, output)
+
+  with tarfile.open(os.path.join(model_dir, "outputs_npy.tgz"), "w:gz") as tar:
+    tar.add(f"{outputs_dir}/", arcname=".")
+  shutil.rmtree(outputs_dir)
+
+
+def _cleanup_hlo(hlo_dir: str, model_dir: str):
+  # The filename of the input HLO varies for each model so we rename it to a
+  # a known name.
+  hlo_files = [
+      f for f in os.listdir(hlo_dir) if re.search(HLO_FILENAME_REGEX, f)
+  ]
+  if len(hlo_files) != 1:
+    raise RuntimeError("Could not find HLO file")
+
+  shutil.move(os.path.join(hlo_dir, hlo_files[0]),
+              os.path.join(model_dir, HLO_STATIC_FILENAME))
+  shutil.rmtree(hlo_dir)
+
+
+def _generate_mlir(jit_function: Any, jit_inputs: Any, model_dir: str,
+                   iree_opt_path: Optional[pathlib.Path]):
+  mlir = jit_function.lower(jit_inputs).compiler_ir(dialect="stablehlo")
+  mlir_path = os.path.join(model_dir, "stablehlo.mlir")
+  print(f"Saving mlir to {mlir_path}")
+  with open(mlir_path, "w") as f:
+    f.write(str(mlir))
+
+  if iree_opt_path:
+    binary_mlir_path = os.path.join(model_dir, "stablehlo.mlirbc")
+    subprocess.run(
+        f"{iree_opt_path} --emit-bytecode {mlir_path} -o {binary_mlir_path}",
+        shell=True,
+        check=True)
+    os.remove(mlir_path)
+
+
+def _generate_artifacts(model: def_types.Model, save_dir: str,
+                        iree_opt_path: pathlib.Path):
+  try:
+    model_dir = os.path.join(save_dir, model.name)
+    os.makedirs(model_dir, exist_ok=True)
```

Comment:
nit (and the same below)
```suggestion
    model_dir.mkdir(parents=True, exist_ok=True)
```